RFS/src/main/java/com/rfs/common/http/ConnectionContext.java

```diff
@@ -24,39 +22,32 @@ public enum Protocol {
     public final Protocol protocol;
     public final String hostName;
     public final int port;
-    public final String username;
-    public final String password;
-    public final AuthType authType;
     public final boolean insecure;
-
-    public ConnectionDetails(Params params) {
-        this(params.getHost(), params.getUsername(), params.getPassword(), params.isInsecure());
+    @Getter
+    private final AuthTransformer authTransformer;
+
+    public ConnectionContext(String url, String username, String password, boolean insecure) {
```

Comment:
Can we pair the constructors down so there is only one that accepts the Params object?  IMO any case  where we are assembling this context manually seems like a bug (except maybe tests)

Reply:
Done